lightning/src/ln/chan_utils.rs

```diff
@@ -497,6 +520,102 @@ impl_writeable_tlv_based!(ChannelPublicKeys, {
+	/// attempt. The tweak is computed as described in [`compute_funding_key_tweak`].
+	//
+	// TODO: Expose `splice_parent_funding_txid` instead so the signer can re-derive the tweak?
+	// There's no harm in the signer trusting the tweak as long as its funding secret has not
+	// been leaked.
+	pub funding_key_tweak: Option<Scalar>,
+}
+
+// `HolderChannelPublicKeys` may have been previously written as `ChannelPublicKeys` so we have to
+// mimic its serialization.
+impl Writeable for HolderChannelPublicKeys {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(0, self.keys.funding_pubkey, required),
+			(2, self.keys.revocation_basepoint, required),
+			(4, self.keys.payment_point, required),
+			(6, self.keys.delayed_payment_basepoint, required),
+			(8, self.keys.htlc_basepoint, required),
+			(10, self.funding_key_tweak, option),
+		});
+		Ok(())
+	}
+}
+
+impl Readable for HolderChannelPublicKeys {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let mut funding_pubkey = RequiredWrapper(None);
+		let mut revocation_basepoint = RequiredWrapper(None);
+		let mut payment_point = RequiredWrapper(None);
+		let mut delayed_payment_basepoint = RequiredWrapper(None);
+		let mut htlc_basepoint = RequiredWrapper(None);
+		let mut funding_key_tweak: Option<Scalar> = None;
+
+		read_tlv_fields!(reader, {
+			(0, funding_pubkey, required),
+			(2, revocation_basepoint, required),
+			(4, payment_point, required),
+			(6, delayed_payment_basepoint, required),
+			(8, htlc_basepoint, required),
+			(10, funding_key_tweak, option),
+		});
+
+		Ok(Self {
+			keys: ChannelPublicKeys {
+				funding_pubkey: funding_pubkey.0.unwrap(),
+				revocation_basepoint: revocation_basepoint.0.unwrap(),
+				payment_point: payment_point.0.unwrap(),
+				delayed_payment_basepoint: delayed_payment_basepoint.0.unwrap(),
+				htlc_basepoint: htlc_basepoint.0.unwrap(),
+			},
+			funding_key_tweak,
+		})
+	}
+}
+
+impl HolderChannelPublicKeys {
+	/// Constructs a new instance of [`HolderChannelPublicKeys`].
+	pub fn new<C: Verification>(
+		funding_pubkey: PublicKey, revocation_basepoint: RevocationBasepoint,
+		payment_point: PublicKey, delayed_payment_basepoint: DelayedPaymentBasepoint,
+		htlc_basepoint: HtlcBasepoint, funding_key_tweak: Option<Scalar>, secp: &Secp256k1<C>,
+	) -> Self {
+		let funding_pubkey = funding_key_tweak
+			.map(|tweak| {
+				funding_pubkey
+					.add_exp_tweak(secp, &tweak)
+					.expect("Addition only fails if the tweak is the inverse of the key")
+			})
+			.unwrap_or(funding_pubkey);
+		Self {
+			keys: ChannelPublicKeys {
+				funding_pubkey,
+				revocation_basepoint,
+				payment_point,
+				delayed_payment_basepoint,
+				htlc_basepoint,
+			},
+			funding_key_tweak,
+		}
+	}
+
+	/// Returns the holder's channel public keys as a [`ChannelPublicKeys`].
+	pub fn inner(&self) -> &ChannelPublicKeys {
+		&self.keys
+	}
```

Comment:
Could maybe implement `AsRef<ChannelPublicKeys>`? Though I guess that only helps when you need the entire struct, not just one of its fields.